source/extensions/filters/network/rbac/rbac_filter.cc

```diff
@@ -13,14 +14,47 @@ namespace Extensions {
 namespace NetworkFilters {
 namespace RBACFilter {
 
+absl::Status ActionValidationVisitor::performDataInputValidation(
+    const Envoy::Matcher::DataInputFactory<Filters::Common::RBAC::Matching::MatchingData>&,
+    absl::string_view type_url) {
+  static absl::flat_hash_set<std::string> allowed_inputs_set{
+      {TypeUtil::descriptorFullNameToTypeUrl(
+          envoy::extensions::matching::common_inputs::network::v3::DestinationIPInput::descriptor()
+              ->full_name())},
+      {TypeUtil::descriptorFullNameToTypeUrl(envoy::extensions::matching::common_inputs::network::
+                                                 v3::DestinationPortInput::descriptor()
+                                                     ->full_name())},
+      {TypeUtil::descriptorFullNameToTypeUrl(
+          envoy::extensions::matching::common_inputs::network::v3::SourceIPInput::descriptor()
+              ->full_name())},
+      {TypeUtil::descriptorFullNameToTypeUrl(
+          envoy::extensions::matching::common_inputs::network::v3::SourcePortInput::descriptor()
+              ->full_name())},
+      {TypeUtil::descriptorFullNameToTypeUrl(
+          envoy::extensions::matching::common_inputs::network::v3::DirectSourceIPInput::descriptor()
+              ->full_name())},
+      {TypeUtil::descriptorFullNameToTypeUrl(
+          envoy::extensions::matching::common_inputs::network::v3::ServerNameInput::descriptor()
+              ->full_name())}};
+  if (allowed_inputs_set.contains(type_url)) {
+    return absl::OkStatus();
+  }
+
+  return absl::InvalidArgumentError(
+      fmt::format("RBAC network filter cannot match on HTTP inputs, saw {}", type_url));
+}
+
 RoleBasedAccessControlFilterConfig::RoleBasedAccessControlFilterConfig(
     const envoy::extensions::filters::network::rbac::v3::RBAC& proto_config, Stats::Scope& scope,
+    Server::Configuration::ServerFactoryContext& context,
     ProtobufMessage::ValidationVisitor& validation_visitor)
     : stats_(Filters::Common::RBAC::generateStats(proto_config.stat_prefix(),
                                                   proto_config.shadow_rules_stat_prefix(), scope)),
       shadow_rules_stat_prefix_(proto_config.shadow_rules_stat_prefix()),
-      engine_(Filters::Common::RBAC::createEngine(proto_config, validation_visitor)),
-      shadow_engine_(Filters::Common::RBAC::createShadowEngine(proto_config, validation_visitor)),
+      engine_(Filters::Common::RBAC::createEngine(proto_config, context, validation_visitor,
+                                                  validation_visitor_)),
+      shadow_engine_(Filters::Common::RBAC::createShadowEngine(
+          proto_config, context, validation_visitor, validation_visitor_)),
```

Comment:
the `validation_visitor` and `validation_visitor_` looks very similar, could we rename to distinguish them a bit?

Reply:
Sure, renamed to `action_validation_visitor_`.